src/Model/Repository/MyParcelConsignmentRepository.php

```diff
@@ -281,7 +283,12 @@ public function setPickupAddressFromCheckout($checkoutData)
             ->setPickupStreet($aCheckoutData['street'])
             ->setPickupCity($aCheckoutData['city'])
             ->setPickupNumber($aCheckoutData['number'])
-            ->setPickupLocationName($aCheckoutData['location']);
+            ->setPickupLocationName($aCheckoutData['location'])
+            ->setPickupLocationCode($aCheckoutData['location_code']);
+
+        if (isset($aCheckoutData['retail_network_id'])) {
+            $aCheckoutData->setPickupNetworkId($aCheckoutData['retail_network_id']);
```

Comment:
The method setPickupNetworkId cannot be called on $aCheckoutData (of type array<string,?,{"price_c...etail_network_id":"?"}>).